infrastructure/src/main/java/org/corfudb/infrastructure/management/FailureDetector.java

```diff
@@ -99,6 +103,10 @@ public PollReport poll(
         );
     }
 
+    private void registerTimerForNodeIfNeeded(String node) {
+        MeterRegistryProvider.getInstance().ifPresent(registry -> registry.timer("failure-detector.ping-latency", "node", node));
```

Comment:
It seems to me you can get rid of some boilerplate code if you add a "timer" method into MetricRegistryProvider
```
public static void timer(String timerName, String tag, String tag2) {
        getInstance()
                .ifPresent(registry -> registry.timer(timerName, tag, tag2));
    }
```

then you can use it:
```
MeterRegistryProvider.timer("failure-detector.ping-latency", "node", node);
```

Reply:
Done